### PR TITLE
More updates to release plan

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Comment by IanMatthewHuff** [comment]
@DonJayamanne  and @joyceerhl  in particular since ya'll were releasing with me and Joyce is releasing next. See if the updates here look like they explain the new process changes well.

File: .github/release_plan.md [hunk 1]
```diff
@@ -56,11 +56,19 @@
...
 
 ## Release
 
 -   [ ] Publish the release
-    -   [ ] For manual
+    -   [ ] For an automated release
+        -   [ ] Create a commit which contains the words `publish` and `release` in it (you can use --allow-empty if needed)
+        -   [ ] Directly push (PR not required) the commit to the `release-xxxx.xx` branch
+        -   [ ] This commit will trigger the `release` stage to run after smoke tests. [Example run](https://github.com/microsoft/vscode-jupyter/actions/runs/702919634)
+        -   [ ] For release branches a mail will be sent to verify that the release should be published. Click the `Review pending deployments` button on the mail and deploy from the GitHub page. This will publish the release on the marketplace.
```
Comment (joyceerhl) [inline .github/release_plan.md:68]: Have we updated the list of people who get this mail?

Comment (IanMatthewHuff) [reply .github/release_plan.md:68]: @joyceerhl we hadn't, but now we have. I've added ya for the next release.

**Review by joyceerhl — APPROVED** [review]
Thanks for updating the release plan 😊

**Pull request merged**